packages/@aws-cdk/aws-lambda/lib/function-base.ts

```diff
@@ -273,6 +273,13 @@ export abstract class FunctionBase extends Resource implements IFunction, ec2.IC
    * @param permission The permission to grant to this Lambda function. @see Permission for details.
    */
   public addPermission(id: string, permission: Permission) {
+    this.addPermissionHelper(id, permission, false);
```

Comment:
Passing around boolean literals here makes it kindof hard to decipher what is going on. I think it would be more readable to include a protected function in the base class called something like `addAnnotationsForPermissionsIfNeeded` (sorry, awfully long name).  The default implementation would include the Annotation. Then the Qualified Function can override it to do nothing. (This is basically the [template pattern](https://en.wikipedia.org/wiki/Template_method_pattern))

Reply:
Great suggestion and thanks for the link to the pattern. I've updated the code to do this instead.